### DISCUSSION TITLE
How should I go about updating the maps?

### DISCUSSION THREAD

**Discussion by eugen-nw** [valhalla/valhalla#3923] [Q&A] [answered]
Hello,

I do not seem to be able to find this piece of information in the docs.  I have a Linux installation of Valhalla dating back to late 2020 and would like to update the maps being used.  How would I accomplish that please?

Thank you,  eugen

**Comment by nilsnolde** [comment, marked as answer]
You just run `valhalla_build_tiles` again (plus the other commands which are important for you) with a current OSM extract of your choice, e.g. from https://download.geofabrik.de/. Then start the service again. There's more info here: https://valhalla.github.io/valhalla/building/#running-valhalla-server-on-unix.

**Comment by eugen-nw** [comment]
Thanks very much!  I'm a novice Linux user and even less knowledgeable with Valhalla.  It'd been ages since I set up this server.  I'm running Valhalla server using the command line "valhalla_service  ~/src/valhalla/valhalla.json  2"

After reading the https://valhalla.github.io/valhalla/building/#running-valhalla-server-on-unix, I can find the /datadrive/valhalla2/valhalla_tiles.tar file, so that might have been the path I took in the past.   Am I correct to guess that if I'd build the index with the `valhalla_build_extract` command the routes would load faster?  Where can I find that command, as `find /datadrive -type f -iname "valhalla_build_extract*"` returns no results?

**Reply by nilsnolde** [reply]
I'd like to help but I honestly think you'd be best off asking someone at work or a buddy. Best advice I can give is: nuke the whole server and start from scratch. Either with building Valhalla from source (as you did before) or e.g. https://github.com/gis-ops/docker-valhalla.

**Reply by eugen-nw** [reply]
Thank you.  It may have happened that the valhalla_build_extract came into existence past late 2021 when I built up our OSM Routing Server.

For anyone who's in my camp and needs to update their Routes, please see below the notes I took in the process.  The update works and the Routes are getting computed.

![image](https://user-images.githubusercontent.com/15199814/214369979-45466bce-a3a3-4a50-b9df-52fd27669d8d.png)
